M/MONA/build_tarballs.jl

```diff
@@ -0,0 +1,52 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "MONA"
+version = v"1.4.18"
+
+# Collection of sources required to complete build
+sources = [
+    ArchiveSource("http://www.brics.dk/mona/download/mona-1.4-18.tar.gz", "ece10e1e257dcae48dd898ed3da48f550c6b590f8e5c5a6447d0f384ac040e4c")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/mona*
+./configure --prefix=${prefix} --build=${MACHTYPE} --host=${target}
+make -j${nproc}
+make install
+install_license COPYING
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
+    Platform("i686", "linux"; libc = "glibc"),
+    Platform("x86_64", "linux"; libc = "glibc"),
+    Platform("aarch64", "linux"; libc = "glibc"),
+    Platform("armv6l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    Platform("powerpc64le", "linux"; libc = "glibc"),
+    Platform("i686", "linux"; libc = "musl"),
+    Platform("x86_64", "linux"; libc = "musl"),
+    Platform("aarch64", "linux"; libc = "musl"),
+    Platform("armv6l", "linux"; call_abi = "eabihf", libc = "musl"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "musl"),
+    Platform("x86_64", "freebsd"; ),
+    Platform("aarch64", "freebsd"; )
+]
+
```

Comment:
```suggestion
platforms = supported_platforms(; exclude=Sys.iswindows)
```

Reply:
Since the suggested fix for apple did not work, I will also exlude it. Should I use this instead?
```suggestion
platforms = supported_platforms(; exclude=(() -> Sys.isapple() | Sys.iswindows()))
```

Reply:
What do you mean? It did work on x86_64 Darwin

Reply:
But it failed on aarch64-apple-darwin

Reply:
1. it isn't clear to me why you want to exclude a platform which does work
2. the problem you see on aarch64-darwin is pretty much the same issue as the one on all [linux platforms](https://github.com/JuliaPackaging/Yggdrasil/pull/9440#issuecomment-2353758120), the difference being that the macOS linker doesn't allow undefined references, while the Linux one by default does, trading off a linker error with a runtime error, but the root is pretty much the same and you want to fix it anyways.